### PR TITLE
Python: improved text_search folder testing

### LINKED ISSUE #9485: Python: Vector store & collection specific exception types
We have the new vector store and collection interfaces. However, we are still using the old memory connector exceptions. We should update the exception types to `VectorStoreXXXException`.

### PR DESCRIPTION
**Pull request by eavanvalkenburg**
### Motivation and Context

added unit tests to up the test coverage across the data folder
Also redoes exceptions for everything vector.

This does introduce some breaking changes on the exceptions returned by the different methods for vector stores, they are still marked experimental and this will go a long way in moving towards release.

Closes #9485 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/9edfaa2e6b5db4f8f010ce66449211dd9ed544aa/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>16732</b></td><td><b>1827</b></td><td><b>89%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 2966 | 4 :zzz: | 0 :x: | 0 :fire: | 1m 9s :stopwatch: |

**Review by moonbox3 — APPROVED** [review]

File: python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
 from collections.abc import Sequence
 from typing import Any, ClassVar, Generic, TypeVar
 
+from semantic_kernel.exceptions.vector_store_exceptions import VectorSearchExecutionException
```
Comment (TaoChenOSU) [inline python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py:9]: nit: group this with the rest of the SK imports

File: python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py [hunk 2]
```diff
@@ -91,7 +96,7 @@ def __init__(
             if not collection_name:
                 collection_name = search_client._index_name
             elif search_client._index_name != collection_name:
-                raise MemoryConnectorInitializationError(
+                raise VectorStoreInitializationException(
```
Comment (TaoChenOSU) [inline python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py:99]: Question: Is this a breaking change? If someone was catching this specific exception before, it's now all of a sudden not catching it.

File: python/tests/integration/completions/test_text_completion.py [hunk 3]
```diff
@@ -2,9 +2,12 @@
 
 import platform
 import sys
-from functools import partial, reduce
+from functools import partial
 from typing import Any
 
+from semantic_kernel.connectors.ai.text_completion_client_base import TextCompletionClientBase
```
Comment (TaoChenOSU) [inline python/tests/integration/completions/test_text_completion.py:8]: nit: group this with the rest of SK imports.

File: python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py (hunk 2 above)
Comment (eavanvalkenburg) [reply python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py:98]: Unfortunately this does mean breaking changes, I've updated the description of this PR.

File: python/semantic_kernel/data/text_search/utils.py [hunk 4]
```diff
@@ -44,30 +48,24 @@ def create_options(
...
+            if inputs:
+                new_options = options_class.model_validate(inputs)
         else:
             new_options = options
         for key, value in kwargs.items():
             if key in new_options.model_fields:
                 setattr(new_options, key, value)
     else:
-        try:
+        with suppress(ValidationError):
```
Comment (TaoChenOSU) [inline python/semantic_kernel/data/text_search/utils.py:67]: Question: With this suppressed, will `new_options` become `None` or something else when an exception happens?

**Review by TaoChenOSU — APPROVED** [review]

File: python/semantic_kernel/data/text_search/utils.py (hunk 4 above)
Comment (eavanvalkenburg) [reply python/semantic_kernel/data/text_search/utils.py:67]: It is already defined earlier as a new object

**Review by moonbox3 — APPROVED** [review]

**Pull request merged**